acra-censor/handlers/handlers_util.go

```diff
@@ -174,17 +174,49 @@ func handleSelectWherePattern(queryNodes, patternNodes []sqlparser.SQLNode) bool
 
 // handle SELECT a, b FROM t1 WHERE userID=%%VALUE%% pattern
 func handleValuePattern(queryNodes, patternNodes []sqlparser.SQLNode) bool {
+	patternNodeOffset := 0
+	queryNodeOffset := 0
+
 	matchDetected := false
-	if len(patternNodes) != len(queryNodes) {
-		return false
-	}
-	for index, patternNode := range patternNodes {
-		if index == 0 || reflect.DeepEqual(patternNode, queryNodes[index]) {
+	for index := 1; index < len(patternNodes); index++ {
+		//This means that checked query nodes are equal to pattern and no more nodes remained, so query matches pattern (no matter if pattern has remained nodes)
+		if index+queryNodeOffset >= len(queryNodes) {
+			break
+		}
+		//This means that checked query nodes are equal to pattern but some more nodes remained, so query doesn't match pattern
+		if index+patternNodeOffset >= len(patternNodes) {
+			return false
+		}
+		//Start check matching
+		if reflect.DeepEqual(patternNodes[index+patternNodeOffset], queryNodes[index+queryNodeOffset]) {
 			continue
 		}
-		if patternNodeComparison, ok := patternNode.(*sqlparser.ComparisonExpr); ok && patternNodeComparison != nil {
-			if queryNodeComparison, ok := queryNodes[index].(*sqlparser.ComparisonExpr); ok && queryNodeComparison != nil {
-				if reflect.DeepEqual(queryNodeComparison.Left, patternNodeComparison.Left) {
+		//handle '*' case
+		if patternSelectExprs, ok := patternNodes[index+patternNodeOffset].(sqlparser.SelectExprs); ok {
+			if _, ok := queryNodes[index+queryNodeOffset].(sqlparser.SelectExprs); ok {
+				if starFound(patternSelectExprs) {
+					for i := index; i < len(queryNodes); i++ {
+						if _, ok := queryNodes[i].(sqlparser.TableExprs); ok {
+							break
+						}
+						queryNodeOffset++
+					}
+					for i := index; i < len(patternNodes); i++ {
+						if _, ok := patternNodes[i].(sqlparser.TableExprs); ok {
+							break
+						}
+						patternNodeOffset++
+					}
+					continue
+				}
+			}
+		}
+
+		if patternNodeComparison, ok := patternNodes[index + patternNodeOffset].(*sqlparser.ComparisonExpr); ok && patternNodeComparison != nil {
+			if queryNodeComparison, ok := queryNodes[index + queryNodeOffset].(*sqlparser.ComparisonExpr); ok && queryNodeComparison != nil {
+				if reflect.DeepEqual(queryNodeComparison.Left, patternNodeComparison.Left) &&
```

Comment:
reflect.DeepEqual don't compare all fields? why not just `reflect.DeepEqual(queryNodeComparison, patternNodeComparison)`? If it didn't, so I suggest to move this logic to some small function and use it to increase readability and understanding:
```
func IsEqualComparisonNode(node1, node2) bool {
  ....
}
```
then there use just
```
  if IsEqualComparisonNode(queryNodeComparison, patternNodeComparison) {
     ....
  } else {
     ....
  }
```
After this refactoring other people (like me) wouldn't need to know about node implementations and their structure to understand what actually we expect and check in this context .